### DISCUSSION TITLE
issue: value prop in register method doesnt work?

### DISCUSSION THREAD

**Discussion by danielhammerl** [react-hook-form/react-hook-form#7226] [Q&A]
### Version Number

7.21.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-js-forked-xifrt?file=/src/App.js

### Steps to reproduce

1. Register a field using the register value and pass down a value: "register('xyz', {value: 'any'});
2. watch to the field or take a look to the react hook form devtools
3. You see no value

### Expected behaviour

Shouldn't passing down a value to the register function return the value if I watch the field?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> Set up value for the registered input. This prop should be utilised inside useEffect or invoke once, each re-run will update or overwrite the input value which you have supplied.

https://react-hook-form.com/api/useform/register

<img width="938" alt="Screen Shot 2021-12-07 at 9 07 48 am" src="https://user-images.githubusercontent.com/10513364/144930543-46dc1114-5ddf-4eca-8405-0d0f6f1185c2.png">

every render when the function gets invoked again, the value will be reset back to the value which you supplied, it's really just a funcition.

**Reply by danielhammerl** [reply]
so that means it will only have that given value on the first re-render, but not initial?

**Reply by bluebill1049** [reply]
you can almost treat it as a hidden input.

**Reply by bluebill1049** [reply]
if you need to update value, you should still use the onChange method which it returns or setValue.

**Reply by danielhammerl** [reply]
ok but i dont change the value, I only register a field with value x. But the input doesnt seem to have value x. Also the documentation says "each re-run will update or overwrite the input value ...", but it doesnt seem so. My input doesnt have any value.

**Reply by bluebill1049** [reply]
> My input doesnt have any value.

`register("test", { value: "ok" });` isn't this a value?
<img width="560" alt="Screen Shot 2021-12-08 at 9 11 41 am" src="https://user-images.githubusercontent.com/10513364/145114013-bcbe645e-89c0-48bc-8d52-e2692d58ca00.png">

**Reply by danielhammerl** [reply]
Yeah i pass a value but the watch  in line 12 doesnt return „ok“. It returns nothing. Is this intentional?

**Reply by bluebill1049** [reply]
that's intentional, so before the form is mounted, `watch` will return only `defaultValues` or inline `defaultValue`

https://react-hook-form.com/api/useform/watch

> When defaultValue is not defined, the first render of watch will return undefined because it is called before register, but you can set the defaultValue as the second argument or provide defaultValues via useForm to avoid this behaviour.

**Reply by danielhammerl** [reply]
ok I understand, but why devtools also dont show any value?
Also if you would take look to the code sandbox again, i changed it a little bit, so now the console.log is in another component inside the form. But it returns still undefined: is this intentional?

**Reply by ljosberinn** [reply]
calling `register(key, { value : 'something' })` should at least show up in the devtools shouldnt it? that it doesnt trigger a rerender seems fair to me considering `setValue` is available in case you need one. but currently the devtools dont pick the value change up either

**Reply by r1ckyrockz** [reply]
Yep, the devtools should update the value as well. i don't want to use `setValue` if `register` (should) should work `register('someName', {value: 'someValue'})` as expected.
